tutorials/v7/climate/climate_tutorial.C

```diff
@@ -0,0 +1,249 @@
+   auto model = RNTupleModel::Create();
+   // To define the data model, create fields with a given C++ type and name.  Fields are roughly TTree branches.
+   // MakeField returns a shared pointer to a memory location to fill the ntuple with data
+   auto fieldYear       = model->MakeField<std::uint32_t>("Year");
+   auto fieldMonth      = model->MakeField<std::uint32_t>("Month");
+   auto fieldDay        = model->MakeField<std::uint32_t>("Day");
+   auto fieldAvgTemp    = model->MakeField<float>("AverageTemperature");
+   auto fieldTempUncrty = model->MakeField<float>("AverageTemperatureUncertainty");
+   auto fieldCity       = model->MakeField<std::string>("City");
+   auto fieldCountry    = model->MakeField<std::string>("Country");
+   auto fieldLat        = model->MakeField<float>("Latitude");
+   auto fieldLong       = model->MakeField<float>("Longitude");
+
+   // Hand-over the data model to a newly created ntuple of name "globalTempData", stored in kNTupleFileName
+   // In return, get a unique pointer to a fillable ntuple (first compress the file)
+   RNTupleWriteOptions options;
+   options.SetCompression(ROOT::RCompressionSetting::EDefaults::kUseGeneralPurpose);
+   auto ntuple = RNTupleWriter::Recreate(std::move(model), "GlobalTempData", kNTupleFileName, options);
+
+   auto file = RRawFile::Create(kRawDataUrl);
+   std::string record;
+   constexpr int kMaxCharsPerLine = 128;
+   while (file->Readln(record)) {
+      if (record.length() >= kMaxCharsPerLine)
+         throw std::runtime_error("record too long: " + record);
+
+      // Parse lines of the form
+      // 1743-11-01,6.068,1.7369999999999999,Århus,Denmark,57.05N,10.33E
+      // and skip records with empty fields
+      std::replace(record.begin(), record.end(), ',', ' ');
+      char country[kMaxCharsPerLine];
+      char city[kMaxCharsPerLine];
+      int nFields = sscanf(record.c_str(), "%u-%u-%u %f %f %s %s %fN %fE",
+                           fieldYear.get(), fieldMonth.get(), fieldDay.get(),
+                           fieldAvgTemp.get(), fieldTempUncrty.get(), country, city,
+                           fieldLat.get(), fieldLong.get());
+      if (nFields != 9) {
+         nSkipped++;
+         continue;
+      }
+      *fieldCountry = country;
+      *fieldCity = city;
+
+      ntuple->Fill();
+
+      if (++nRecords % 1000000 == 0)
+         std::cout << "  ... converted " << nRecords << " records" << std::endl;
+   }
+
+   // Display the total time to process the data
+   std::cout << nSkipped << " records skipped" << std::endl;
+   std::cout << nRecords << " records processed" << std::endl;
+
+   auto t2 = Clock::now();
+   std::cout << std::endl
+             << "Processing Time: "
+             << std::chrono::duration_cast<std::chrono::seconds>(t2 - t1).count()
+             << " seconds\n" << std::endl;
+}
+
+
+void Analyze() {
+   // Get a unique pointer to an empty RNTuple model
+   auto model = RNTupleModel::Create();
```

Comment:
Let's skip model definition altogether.  RDataFrame does the modeling internally, so we only need it to print one of the entries.  We can simply do this on the complete model.  To do so, we remove the model and call

    auto ntuple = RNTupleReader::Open("GlobalTempData", kNTupleFileName);